Sources/SwiftBSON/BSONDocumentIterator.swift

```diff
@@ -95,95 +94,173 @@ public class BSONDocumentIterator: IteratorProtocol {
         return bsonType
     }
 
+    /// Search for the value associated with the given key, returning its type if found and nil otherwise.
+    /// This moves the iterator right up to the first byte of the value.
+    internal func findValue(forKey key: String) -> BSONType? {
```

Comment:
could we possibly match by accident on the right string but in the wrong place? like if a subdocument contains the same key, or a string value contains it, etc.

Reply:
I don't think so, since we use `skipNextValue` between each attempt to read a key.